src/core/store.ts

```diff
@@ -335,15 +335,23 @@ export const createStore = (
         cancelSuspensePromise(atomState.p)
       }
     }
+
+    const { d, changed: dependenciesChanged } = createReadDependencies(
+      version,
+      atomState?.d,
+      dependencies
+    )
+
     const nextAtomState: AtomState<Value> = {
       v: value,
       r: atomState?.r || 0,
-      d: createReadDependencies(version, atomState?.d, dependencies),
+      d,
     }
     if (
       !atomState ||
       !('v' in atomState) || // new value, or
-      !Object.is(atomState.v, value) // different value
+      !Object.is(atomState.v, value) || // different value,
+      dependenciesChanged
```

Comment:
Updates the revision if the dependencies have changed. That's necessary to track when the dependent atom has the same initial value as the accessed one.